cmd/xl-v1-utils.go

```diff
@@ -42,14 +41,45 @@ func reduceErrs(errs []error, ignoredErrs []error) error {
 		errorCounts[err]++
 	}
 	max := 0
-	var errMax error
 	for err, count := range errorCounts {
 		if max < count {
 			max = count
-			errMax = err
+			maxErr = err
+		}
+	}
+	return max, maxErr
+}
+
+// reduceReadQuorumErrs behaves like reduceErrs but only for returning
+// values of maximally occuring errors validated against readQuorum.
+func reduceReadQuorumErrs(errs []error, ignoredErrs []error, readQuorum int) (maxErr error) {
+	maxCount, maxErr := reduceErrs(errs, ignoredErrs)
+	if maxCount >= readQuorum {
+		if maxErr != nil {
```

Comment:
While this is correct, we can simplify as follows. There is no need to handle ``nil`` differently.
```go
if maxCount >= readQuorum {
  return traceError(maxErr, errs)
} else {
  return traceError(errXLReadQuorum, errs)
}
```